pywbem/mof_compiler.py

```diff
@@ -2089,14 +2096,21 @@ def CreateClass(self, *args, **kwargs):
         cc = args[0] if args else kwargs['NewClass']
         if cc.superclass:
             try:
-                _ = self.GetClass(cc.superclass, LocalOnly=True,  # noqa: F841
-                                  IncludeQualifiers=False)
+                # Since this may cause recursive GetClass calls
+                # IncludeQualifiers = True to insure reference properties on
+                # instances with aliases get built correctly.
+                self.GetClass(cc.superclass, LocalOnly=True,
+                              IncludeQualifiers=True)
             except CIMError as ce:
                 if ce.status_code == CIM_ERR_NOT_FOUND:
                     ce.args = (CIM_ERR_INVALID_SUPERCLASS, cc.superclass)
                     raise
                 raise
 
+        # TODO: Not sure why this code puts cc in local repo here and then
+        # again at the end. Note that it executes tests on the class
+        # before again inserting it into the repo and these can cause
+        # exceptions.
```

Comment:
The one at the end updates `self.class_names`, while this here updates `self.classes`.

I think that both should be at the end, since I don't think we want any updates to the repo in case of raising exceptions.

Reply:
I also questioned this one which is why I added the comment but it has been there for a long time and this code is convoluted enough that I am uncomfortable because I am not sure if the subsequent  code in mp_createclass that could repeatedly do GetClass, find_mof and compile new mof to the cache might be caught in something I don't understand.

On the other hand a failure here causes the process to stop so nothing is really lost. Further, rollback starts with the class_names dictionary to get classes.

The only thing I can suggest here is to try moving that code  and see what happens.

Reply:
You have added code to clean up the added objects in case an exception is raised. That is good enough from my perspective, for now.